build-system/tasks/helpers.js

```diff
@@ -460,6 +460,12 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
               continue;
             }
             const isExternal = external.includes(value);
+            if (value === 'react' || value === 'react-dom') {
+              return {
+                path: isExternal ? value : require.resolve(value),
```

Comment:
why is this branch necessary? `require.resolve('react')` should return the same path as `resolvePath('react')`, right? (and same for `'react-dom'`

Reply:
It doesn't for some reason. So I had to write this hacky code to make it work. I thought I'd revisit it later to see if I couldn't make it work better.

Reply:
I get this error without using `require.resolve`

```
node_modules/esbuild/lib/main.js:250:10: error: [plugin: remap-dependencies] "path" must be a string
```

Reply:
that's probably because `resolvePath()` is returning `null` and esbuild is complaining that it needs `path` to be a `string`. From prior experience, resolvePath does that when it can't find the module (in node_modules), but that's odd since they should both be there?